db/sql/test_sql_functions.sql

```diff
@@ -298,6 +298,123 @@ END;
 $f$ LANGUAGE plpgsql;
 
 
+-- msar.prepare_temp_table_for_import -------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION test_msar_prepare_temp_table_for_import() RETURNS SETOF TEXT AS $f$
+DECLARE
+  r_json jsonb;
+  table_oid bigint;
+BEGIN
+  r_json := msar.prepare_temp_table_for_import(
+    tab_name := 'tmp_test_table',
+    col_names := ARRAY['id','other']
+  );
+  table_oid := (r_json->>'table_oid')::bigint;
+  RETURN NEXT ok(
+    table_oid IS NOT NULL,
+    'prepare_temp_table_for_import returns table_oid'
+  );
+
+  r_json := msar.prepare_temp_table_for_import(
+    tab_name := 'tmp_explicit_table',
+    col_names := ARRAY['id','other']
+  );
+  table_oid := (r_json->>'table_oid')::bigint;
+  RETURN NEXT ok(
+    (SELECT relpersistence = 't' FROM pg_class WHERE oid = table_oid),
+    'explicit tab_name created as temporary table'
+  );
+
+  r_json := msar.prepare_temp_table_for_import(
+    tab_name := 'tmp_copy_check',
+    col_names := ARRAY['id','other']
+  );
+  RETURN NEXT matches(
+    r_json->>'copy_sql',
+    '^COPY ',
+    'copy_sql returned and starts with COPY'
+  );
+END;
+$f$ LANGUAGE plpgsql;
+
+
+-- msar.insert_from_select -----------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION __setup_msar_insert_from_select() RETURNS SETOF TEXT AS $$
+BEGIN
+  CREATE TEMPORARY TABLE tmp_src_insert (
+    text_col text,
+    num_col text,
+    bool_col text,
+    numeric_col text,
+    email_col text,
+    money_col text
+  );
+  INSERT INTO tmp_src_insert VALUES 
+    ('100', '10', 'true', '123.45', 'test@example.com', '$50.00'),
+    ('200', '20', 'false', '678.90', 'user@test.org', '$100.50');
+  
+  CREATE TABLE insert_dest_table (
+    id integer,
+    value integer,
+    is_active boolean,
+    amount numeric,
+    email mathesar_types.email,
+    price mathesar_types.mathesar_money
+  );
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION test_msar_insert_from_select_success() RETURNS SETOF TEXT AS $f$
+BEGIN
+  PERFORM __setup_msar_insert_from_select();
+
+  RETURN NEXT lives_ok($$
+    SELECT msar.insert_from_select(
+      'tmp_src_insert'::regclass,
+      'insert_dest_table'::regclass,
+      jsonb_build_array(
+        jsonb_build_object('src_table_attnum', 1, 'dst_table_attnum', 1),
+        jsonb_build_object('src_table_attnum', 2, 'dst_table_attnum', 2),
+        jsonb_build_object('src_table_attnum', 3, 'dst_table_attnum', 3),
+        jsonb_build_object('src_table_attnum', 4, 'dst_table_attnum', 4),
+        jsonb_build_object('src_table_attnum', 5, 'dst_table_attnum', 5),
+        jsonb_build_object('src_table_attnum', 6, 'dst_table_attnum', 6)
```

Comment:
Can you shuffle the `src` table columns such that the mappings aren't one-one i.e. make it so that e.g. column 3 of `src` goes to column 1 of `dst`, and so on.